janusgraph-core/src/main/java/org/janusgraph/graphdb/transaction/StandardJanusGraphTx.java

```diff
@@ -866,18 +868,18 @@ private void connectRelation(InternalRelation r) {
     }
 
     public JanusGraphVertexProperty addProperty(JanusGraphVertex vertex, PropertyKey key, Object value) {
-        return addProperty(vertex, key, value, null);
+        return addProperty(vertex, key, value, null, 0L);
     }
 
-    public JanusGraphVertexProperty addProperty(JanusGraphVertex vertex, PropertyKey key, Object value, Long id) {
-        return addProperty(key.cardinality().convert(), vertex, key, value, id);
+    public JanusGraphVertexProperty addProperty(JanusGraphVertex vertex, PropertyKey key, Object value, Long id, long previousId) {
+        return addProperty(key.cardinality().convert(), vertex, key, value, id, previousId);
     }
 
     public JanusGraphVertexProperty addProperty(VertexProperty.Cardinality cardinality, JanusGraphVertex vertex, PropertyKey key, Object value) {
-        return addProperty(cardinality, vertex, key, value, null);
+        return addProperty(cardinality, vertex, key, value, null, 0L);
     }
 
-    public JanusGraphVertexProperty addProperty(VertexProperty.Cardinality cardinality, JanusGraphVertex vertex, PropertyKey key, Object value, Long id) {
+    public JanusGraphVertexProperty addProperty(VertexProperty.Cardinality cardinality, JanusGraphVertex vertex, PropertyKey key, Object value, Long id, long previousId) {
```

Comment:
I prefer `Long` to `long`. Three reasons: 1) In the future, we might want to support property id of other non-primitive types. 2) Existing code is using `Long`. 3) Using "0" as "undefined" is not a well-established contract in JanusGraph.